src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +162,171 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
+            bool validTransactionDescribeAcl =
+                transactionIdsDescribeAclBindings.Any(transactionIdsAclBinding => AssertAclMatchingName(TransactionalId, transactionIdsAclBinding));
+            bool validTopicWriteAcl = topicsWriteAclBindings.Any(topicsAclBinding => AssertAclMatchingName(topicName, topicsAclBinding));
+
+            if (validTransactionWriteAcl && validTopicWriteAcl && validTransactionDescribeAcl)
             {
-                _producer.AbortTransaction();
+                return false;
             }
-            catch (Exception e)
+
+            var msgBuilder = new StringBuilder($"ETL process: {Name}. Failed to load data for the producer instance. " +
+                                               $"Please check your Kafka security configuration and ensure all of the (RESOURCE : OPERATION) ACL bindings listed below are set to ALLOW: {Environment.NewLine}");
+
+            if ((validTransactionDescribeAcl && validTransactionWriteAcl) == false)
             {
-                if (Logger.IsOperationsEnabled)
-                    Logger.Operations($" ETL process: {Name}. Aborting Kafka transaction failed.", e);
+                msgBuilder.Append($"Transactional ID '{TransactionalId}' : ");
+                if (validTransactionWriteAcl == false)
+                    msgBuilder.Append("WRITE operation ");
+                if (validTransactionDescribeAcl == false)
+                    msgBuilder.Append("DESCRIBE operation");
+                msgBuilder.Append(Environment.NewLine);
             }
 
-            throw new QueueLoadException(ex.Message, ex);
+            
+            if (validTopicWriteAcl == false)
+                msgBuilder.Append($"Topic '{topicName}' : WRITE operation");
+
+            message = msgBuilder.ToString();
+
+        }
+        catch (Exception)
+        {
+            return false;
         }
 
-        return count;
+        return true;
+    }
+
+    private static bool AssertAclMatchingName(string expectedName, AclBinding aclBinding)
+    {
+        return aclBinding.Pattern.ResourcePatternType switch
+        {
+            ResourcePatternType.Literal => aclBinding.Pattern.Name == expectedName,
+            ResourcePatternType.Prefixed => expectedName.StartsWith(aclBinding.Pattern.Name),
+            _ => throw new NotSupportedException()
+        };
+    }
+    
+
+    private void ThrowTransactionalIdAclAuthorizationException(Exception e)
+    {
+        string msg = $"ETL process: {Name}. Failed to initialize transactions for the producer instance. " +
+                     $"Add an required ACL permissions to your API Key that will make WRITE and DESCRIBE operations possible for transactional ID '{TransactionalId}'. ";
+                                    
+        if (Logger.IsOperationsEnabled)
+            Logger.Operations(msg, e);
```

Comment:
Won't it be logged twice?

Reply:
erased https://github.com/ravendb/ravendb/pull/17577/commits/87d9b0d5c8cbe324fcd5d9bb7fb8efcfe53cc977